### PR TITLE
Issue #214 - harvesting form shows wrong tags when deleting request

### LINKED ISSUE #214: harvesting form shows wrong tags when deleting request
How to replicate:

1 Go to https://dev.clearlydefined.io/harvest
1  Add a git repo, select a tag.
1  Add another git repo, select a tag.
1  Delete first (top) repo.

What happens:

Tags for first repo stick around in pulldown, next to second repo's name

What should happen:

Second repo should have second repo's tags.

### PR DESCRIPTION
**Pull request by storrisi**
Fixes #214

### CODE REVIEW

File: src/components/HarvestQueueList.js [hunk 1]
```diff
@@ -53,7 +53,8 @@ class HarvestQueueList extends React.Component {
   commitChanged(request, value) {
     const newRequest = clone(request)
     newRequest.revision = value ? value.sha : null
-    this.setState({ contentSeq: this.state.contentSeq + 1 })
+    newRequest.commit = value
```
Comment (jeffmcaffer) [inline src/components/HarvestQueueList.js]: feels odd to be adding a new property just for this case. Can we just be using `revision`?

File: src/components/HarvestQueueList.js [hunk 2]
```diff
@@ -159,6 +160,7 @@ class HarvestQueueList extends React.Component {
   render() {
     const { loadMoreRows, listHeight, noRowsRenderer, list } = this.props
     const { sortOrder, contentSeq } = this.state
+
```
Comment (jeffmcaffer) [inline src/components/HarvestQueueList.js]: remove the extra blanks here and elsewhere

File: src/components/MavenVersionPicker.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ export default class MavenVersionPicker extends Component {
 
   constructor(props) {
     super(props)
-    this.state = { customValues: [], options: [] }
+    this.state = { customValues: [], options: [], selected: props.request.revision ? [props.request.revision] : [] }
```
Comment (jeffmcaffer) [inline src/components/MavenVersionPicker.js]: All of the changes here and in the other pickers are duplicate. Need some way to either eliminate the need for the code or share it.

Related, it feels like this scenario is not that unusual and ought not be this hard to implement. This approach feels sort of subtle. I don't know a better one right now but if feels overly complicated. (could just be my naivety)

File: src/components/MavenVersionPicker.js [hunk 4]
```diff
@@ -23,6 +23,13 @@ export default class MavenVersionPicker extends Component {
     this.getOptions('')
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.setState(prevState => ({
```
Comment (jeffmcaffer) [inline src/components/MavenVersionPicker.js]: can this be

```javascript
this.setState({...this.state, selected: nextProps.request.revision ? [nextProps.request.revision] : []})
```

That is, is the function needed

File: src/components/HarvestQueueList.js (hunk 1 above)
Comment (storrisi) [reply src/components/HarvestQueueList.js]: The `commit` property is needed for the GitHubCommitPicker, so we had to add it in order to use it

**Comment by storrisi** [comment]
@jeffmcaffer i've updated the PR based on your suggestions

**Pull request merged**